base.go

```diff
@@ -46,28 +46,26 @@ func (b Base) Validate() error {
 func (b Base) String() string {
 	str := strings.ToLower(b.Name)
 	if !b.Channel.Empty() {
-		str += "/" + b.Channel.String()
+		str += "@" + b.Channel.String()
 	}
 	if len(b.Architectures) > 0 {
 		str = fmt.Sprintf("%s on %s", str, strings.Join(b.Architectures, ", "))
 	}
 	return str
 }
 
-// ParseBase parses a base as string in the form "os/track/risk/branch"
+// ParseBase parses a base as string in the form "os@track/risk/branch"
 // and an optional list of architectures
 func ParseBase(s string, archs ...string) (Base, error) {
 	var err error
 	base := Base{}
 
-	// Split the first forward-slash to get name and channel.
-	// E.g. "os/track/risk/branch" => ["os", "track/risk/branch"]
-	segments := strings.SplitN(s, "/", 2)
-	base.Name = strings.ToLower(segments[0])
-	channelName := ""
-	if len(segments) == 2 {
-		channelName = segments[1]
+	segments := strings.Split(s, "@")
+	if len(segments) != 2 {
+		return Base{}, errors.NotValidf("base string must contain exactly one @. %q", s)
```

Comment:
So this isn't correct. You either need to say `@` is always mandatory and you fix `String()` to not return a base without an `@` symbol. Or you need to relax `ParseBase` and if it doesn't have an `@` symbol it just uses the name as the first segment.

Personally, you probably need to fix `String` and you may need to panic if there is a missing `channel`. I believe a panic __may__ be ok here, as `Validate` should ensure that a `channel` should be valid. An alternative may be to just always insert the `@` symbol. Although that will look weird. `ubuntu@` or `ubuntu@/risk/branch`